### DISCUSSION TITLE
OpenAPI model not compliant with anyOf

### DISCUSSION THREAD

**Discussion by rushton** [fastapi/fastapi#8655] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Union, Tuple

from fastapi import FastAPI
from pydantic import BaseModel, StrictFloat, StrictInt

app = FastAPI()


class Foo(BaseModel):
    operand: Union[Tuple[StrictFloat, StrictFloat], Tuple[StrictInt, StrictInt]]


@app.get("/")
def foo(request: Foo):
    return "hello"


app.openapi()
```

### Description

When running `app.openapi()` I get the following error:

```
Traceback (most recent call last):
  File "foo.py", line 19, in <module>
    app.openapi()
  File "/Users/nrushton/Library/Caches/pypoetry/virtualenvs/egress-Y6TME6Xy-py3.8/lib/python3.8/site-packages/fastapi/applications.py", line 136, in openapi
    self.openapi_schema = get_openapi(
  File "/Users/nrushton/Library/Caches/pypoetry/virtualenvs/egress-Y6TME6Xy-py3.8/lib/python3.8/site-packages/fastapi/openapi/utils.py", line 410, in get_openapi
    return jsonable_encoder(OpenAPI(**output), by_alias=True, exclude_none=True)  # type: ignore
  File "pydantic/main.py", line 406, in pydantic.main.BaseModel.__init__
pydantic.error_wrappers.ValidationError: 3 validation errors for OpenAPI
components -> schemas -> Foo -> properties -> operand -> anyOf -> 0 -> items
  value is not a valid dict (type=type_error.dict)
components -> schemas -> Foo -> properties -> operand -> anyOf -> 1 -> items
  value is not a valid dict (type=type_error.dict)
components -> schemas -> Foo -> $ref
  field required (type=value_error.missing)
  ```
  
This became an issue for us when upgrading from `0.65.3` to `0.69`

### Operating System

macOS

### FastAPI Version

0.69

### Python Version

3.8.2

### Additional Context

I delved into the code a little bit and it seems the change from `anyOf: Optional[List[Any]]` to `anyOf: Optional[List["Schema"]]` in `fastapi.openapi.models.Schema` seems to be the cause.

_No response_

**Comment by dconathan** [comment]
Some related issues/instances of the same error:

https://github.com/tiangolo/fastapi/issues/3898
https://github.com/tiangolo/fastapi/issues/3665
https://github.com/tiangolo/fastapi/issues/3782

**Comment by Gingernaut** [comment]
+1 on this issue

**Comment by martinezpl** [comment]
This issue can be closed, the problem doesn't occur in the latest version (0.79.0)
